### PR TITLE
DXP-0003

### PR DESCRIPTION
**Pull request by GetOn4**

### CODE REVIEW

File: aflr/__init__.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 api_key = None
 client_id = None
-api_base = "https://v1.api.audio"
+api_base = "https://api.aflr.io"
```
Comment (tonythree) [inline aflr/__init__.py:10]: this is not correct

File: aflr/api_resources/mastering.py [hunk 2]
```diff
@@ -8,17 +8,32 @@ class Mastering(APIRequest):
     def __init__(self):
         super().__init__()
         self.file_url = self.api_base + "/file/mastering"
+        self.mastering_url = self.api_base + "/mastering"
```
Comment (tonythree) [inline aflr/api_resources/mastering.py:11]: should be `self.url` in order to follow naming conventions with other classes

File: aflr/api_resources/mastering.py [hunk 3]
```diff
@@ -8,17 +8,32 @@ class Mastering(APIRequest):
...
+        if type(backgroundTrackId) != str:
+            raise Exception("backgroundTrackId needs to be a string.")
+        if type(scriptId) != str:
+            raise Exception("scriptId needs to be a string.")
+        body = {
+            "scriptId": scriptId,
+            "backgroundTrackId": backgroundTrackId,
+            "audience": audience,
+        }
+        return self._post_request(url=self.mastering_url, json=body)
```
Comment (tonythree) [inline aflr/api_resources/mastering.py:28]: self.url

File: aflr/api_resources/sound.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        # good read is https://stackoverflow.com/questions/1385759/should-init-call-the-parent-classs-init
+        self.url = self.api_base + "/sound"
+
+    def retrieve(self, scriptId, backgroundTrackId, audience=[]):
+        if type(audience) != list:
+            raise Exception("audience needs to be a list.")
+        if type(backgroundTrackId) != str:
+            raise Exception("backgroundTrackId needs to be a string.")
+        if type(scriptId) != str:
+            raise Exception("scriptId needs to be a string.")
```
Comment (tonythree) [inline aflr/api_resources/sound.py:19]: great checkers. we will need to use pydantic or mypy for type checking. this is good for now

**Review by tonythree — APPROVED** [review]
lgtm

File: aflr/api_resources/mastering.py [hunk 5]
```diff
@@ -8,17 +8,32 @@ class Mastering(APIRequest):
...
+            raise Exception("audience needs to be a list.")
+        if type(backgroundTrackId) != str:
+            raise Exception("backgroundTrackId needs to be a string.")
+        if type(scriptId) != str:
+            raise Exception("scriptId needs to be a string.")
+        body = {
+            "scriptId": scriptId,
+            "backgroundTrackId": backgroundTrackId,
+            "audience": audience,
+        }
```
Comment (zeritte) [inline aflr/api_resources/mastering.py:27]: @GetOn4 I dont think you need to put them into a dictionary. they are put by default. see other post requests

File: aflr/api_resources/sound.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        body = {
+            "scriptId": scriptId,
+            "backgroundTrackId": backgroundTrackId,
+            "audience": audience,
+        }
```
Comment (zeritte) [inline aflr/api_resources/sound.py:24]: the same body comment

File: example.py [hunk 7]
```diff
@@ -1,32 +1,33 @@
 import aflr
 
-aflr.api_key = "your-key"
-
+aflr.api_key = "vyCTlS1hLc5N5N1CON2nh2qtU6PBz7vY1MFfIlS6"
```
Comment (zeritte) [inline example.py]: please delete this api key from the api gateway. there are github bots wandering to collect api keys

**Review by zeritte — CHANGES_REQUESTED** [review]

File: example.py (hunk 7 above)
Comment (tonythree) [reply example.py]: solved and deleted api-key

File: end2end.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import aflr
+import requests
+import time
+
+aflr.api_key = "7pLmUFRgzEYa6Uvbs7qA11wH6hnoxfr856jdoWC8"
```
[the remark this answers is not in the document]
Comment (tonythree) [reply end2end.py]: solved and deleted api key.

**Review by tonythree — APPROVED** [review]
LGTM

**Pull request merged**